### DISCUSSION TITLE
Twitch video support

### DISCUSSION THREAD

**Discussion by jromal** [Lednerb/bilberry-hugo-theme#510] [Ideas]
Hi all.

I was quite surprised to discover there is no Twitch video support.
I added it myself for my local layouts but would be great if it is working out of the box for other people.
Still mine has two problems that are the same.
- It does not render locally, because the value of "parent" is needed. And I have setup it to the webpage domain (maybe there are options to bypass it). 
 
Because it is not rocket science, and even a beginner like me managed it, I do not put here the code unless someone wants it.

But the general question remains. Should not the theme have Twitch support? 
Specially supporting other video sources that far less popular (many I have never heard of them).

**Comment by igor-baiborodine** [comment]
@jromal, for posts of the `article` type you can use raw HTML `iframe` embed to display a Twitch video, but you have to enable [raw HTML](https://github.com/Lednerb/bilberry-hugo-theme#raw-html) in your `config.yml` file. Or you can encapsulate the `iframe` embed within a shortcode. You can read this [article](https://www.kiroule.com/article/use-video-embeds-in-hugo-theme/) for more details.

As per the `video` archetype, I have an idea of how to make it extensible with shortcodes. I will try this in the upcoming `v4` development of the Bilberry Hugo theme.

**Reply by jromal** [reply]
Many thanks. Great article explaining how to embed videos.
